File: November 2024 Leetcode Solution/November-22.cpp
class Solution {
public:
    int maxEqualRowsAfterFlips(vector<vector<int>>& matrix) {
        unordered_map<string, int> count;
        int maxCount = 0;

        for (const auto& row : matrix) {
            string key;
            bool flip = row[0] == 1;

            for (int num : row) {
                key += flip ? ('0' + (1 - num)) : ('0' + num);
            }

            count[key]++;
            maxCount = max(maxCount, count[key]);
        }

        return maxCount;
    }
};

File: November 2024 Leetcode Solution/November-22.rs
use std::collections::HashMap;

impl Solution {
    pub fn max_equal_rows_after_flips(matrix: Vec<Vec<i32>>) -> i32 {
        let mut count = HashMap::new();
        let mut max_count = 0;

        for row in matrix.iter() {
            let flip = row[0] == 1;
            let key: Vec<i32> = row.iter()
                                   .map(|&num| if flip { 1 - num } else { num })
                                   .collect();

            *count.entry(key.clone()).or_insert(0) += 1;
            max_count = max_count.max(*count.get(&key).unwrap());
        }

        max_count
    }
} 
